core/capabilities/fakes/evm_chain.go

```diff
@@ -425,14 +427,50 @@ func (fc *FakeEVMChain) GetTransactionReceipt(ctx context.Context, metadata comm
 func (fc *FakeEVMChain) HeaderByNumber(ctx context.Context, metadata commonCap.RequestMetadata, input *evmcappb.HeaderByNumberRequest) (*commonCap.ResponseAndMetadata[*evmcappb.HeaderByNumberReply], error) {
 	fc.eng.Infow("EVM Chain HeaderByNumber Started", "input", input)
 
-	// Prepare header by number request
-	blockNumber := new(big.Int).SetBytes(input.BlockNumber.AbsVal)
+	var (
+		header *types.Header
+		err    error
+	)
+
+	// Convert the request block number preserving sign.
+	var reqNum *big.Int
+	if input != nil && input.BlockNumber != nil {
+		reqNum = fromProtoBigInt(input.BlockNumber)
```

Comment:
Doesn't this type already have a method to convert to `*big.Int`?
```suggestion
		reqNum = input.BlockNumber.Int()
```
https://pkg.go.dev/github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb#BigInt.Int

Reply:
Oh wrong type, but the one I linked was the original.

Reply:
Ah here: https://github.com/smartcontractkit/chainlink-protos/blob/f8fca0955840976623d1d96078cf1581913139b5/cre/go/values/pb/values.go#L107
```suggestion
		reqNum = pb.NewIntFromBigInt(input.BlockNumber)
```